File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcRepresentationItem/IfcGeometricRepresentationItem/IfcHalfSpaceSolid/IfcPolygonalBoundedHalfSpace/IfcPolygonalBoundedHalfSpace.vb
Public Class IfcPolygonalBoundedHalfSpace Inherits IfcHalfSpaceSolid
    Public Position As IfcAxis2Placement3D
    Public PolygonalBoundary As IfcBoundedCurve

    ' === WHERE CLAUSES ===
    ' BoundaryDim : PolygonalBoundary.Dim = 2
    ' BoundaryType : SIZEOF(TYPEOF(PolygonalBoundary) * [ 'IFC4X3_DEV_73740fe4.IFCPOLYLINE', 'IFC4X3_DEV_73740fe4.IFCCOMPOSITECURVE', 'IFC4X3_DEV_73740fe4.IFCINDEXEDPOLYCURVE' ]) = 1
End Class

File: src/main/resources/output/IFC4X3_python/Entities/IfcRepresentationItem/IfcGeometricRepresentationItem/IfcHalfSpaceSolid/IfcPolygonalBoundedHalfSpace/IfcPolygonalBoundedHalfSpace.py
class IfcPolygonalBoundedHalfSpace(IfcHalfSpaceSolid):
    def __init__(self):
        self.Position: IfcAxis2Placement3D = None
        self.PolygonalBoundary: IfcBoundedCurve = None

    # === WHERE CLAUSES ===
    # BoundaryDim : PolygonalBoundary.Dim = 2
    # BoundaryType : SIZEOF(TYPEOF(PolygonalBoundary) * [ 'IFC4X3_DEV_73740fe4.IFCPOLYLINE', 'IFC4X3_DEV_73740fe4.IFCCOMPOSITECURVE', 'IFC4X3_DEV_73740fe4.IFCINDEXEDPOLYCURVE' ]) = 1
